src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -580,3 +591,18 @@ def get_instance(self, hook_name: str) -> Union["pl.LightningModule", "pl.Lightn
                 " `LightningDataModule`. It will use the implementation from `LightningModule` instance."
             )
         return self.model
+
+
+def _check_dataloader_none(
+    dataloader: Optional[Union[TRAIN_DATALOADERS, EVAL_DATALOADERS]],
+    dataloader_source: _DataLoaderSource,
+    trainer_fn: TrainerFn,
+) -> None:
+    # A prefix in the message to disambiguate between the train- and (optional) val dataloader that .fit() accepts
+    prefix = "training " if trainer_fn == TrainerFn.FITTING else ""
+    if dataloader is None and not dataloader_source.is_defined():
+        raise ValueError(
+            f"No valid {prefix}dataloader was passed to `Trainer.{trainer_fn}()`."
+            f" Either pass the dataloader to the `.{trainer_fn}()` method OR implement"
+            f" `def {dataloader_source.name}(self):` in your LightningModule/LightningDataModule."
```

Comment:
@rohitgr7 Can you explain why you think this is important? It looks to me very unnatural to read.

Reply:
@rohitgr7 I changed it a bit now. I hope this is a good compromise. Personally not 100% happy because it does not generalize well as a message to datamodule users.

Reply:
I was trying to address this: https://github.com/Lightning-AI/lightning/pull/14637#discussion_r969280632

Reply:
Okay, thanks. Are you ok with the current way? Want to make sure your concern was fully resolved.

Reply:
yes, it's fine